compiler/noirc_evaluator/src/ssa/opt/remove_bit_shifts.rs

```diff
@@ -169,14 +177,24 @@ impl Context<'_> {
             let lhs_sign_as_field = self.insert_cast(lhs_sign, NumericType::NativeField);
             let lhs_as_field = self.insert_cast(lhs, NumericType::NativeField);
             // For negative numbers, convert to 1-complement using wrapping addition of a + 1
-            let one_complement = self.insert_binary(lhs_sign_as_field, BinaryOp::Add, lhs_as_field);
+            // Unchecked add as these are fields
+            let one_complement = self.insert_binary(
+                lhs_sign_as_field,
+                BinaryOp::Add { unchecked: true },
+                lhs_as_field,
+            );
             let one_complement = self.insert_truncate(one_complement, bit_size, bit_size + 1);
             let one_complement = self.insert_cast(one_complement, NumericType::signed(bit_size));
             // Performs the division on the 1-complement (or the operand if positive)
             let shifted_complement = self.insert_binary(one_complement, BinaryOp::Div, pow);
             // Convert back to 2-complement representation if operand is negative
             let lhs_sign_as_int = self.insert_cast(lhs_sign, lhs_typ);
-            let shifted = self.insert_binary(shifted_complement, BinaryOp::Sub, lhs_sign_as_int);
+            // TODO: should this be unchecked?
+            let shifted = self.insert_binary(
+                shifted_complement,
+                BinaryOp::Sub { unchecked: false },
+                lhs_sign_as_int,
+            );
```

Comment:
Oh, I think I did the merge wrong. This is the TODO which I wasn't sure about. This is all inside `insert_wrapping_shift_left` so I guess no overflow should happen here, but I'm not sure.

Reply:
We should at least be able to create a testcase for this scenario.